go/networkqualityd.go

```diff
@@ -275,39 +489,38 @@ func setNoPublicCache(h http.Header) {
 
 // slurpHandler reads the post request and returns JSON with bytes
 // read and how long it took
-func slurpHandler(w http.ResponseWriter, r *http.Request) {
+func (h *handlers) slurpHandler(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/octet-stream")
-	setCors((w.Header()))
 	setNoPublicCache(w.Header())
 
-	t := time.Now()
-	n, err := io.Copy(ioutil.Discard, r.Body)
+	if h.enableCORS {
+		setCors((w.Header()))
+	}
+
+	_, err := io.Copy(ioutil.Discard, r.Body)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
-	resp := struct {
-		Duration time.Duration `json:"DurationMS"`
-		Bytes    int64
-		BPS      int64
-	}{
-		Duration: time.Since(t) / time.Millisecond,
-		Bytes:    n,
-	}
+	w.WriteHeader(200)
+}
 
-	if resp.Duration > 0 && resp.Bytes > 0 {
-		resp.BPS = int64(float64(resp.Bytes) / (float64(resp.Duration) / 1000))
+// ignorableError returns true if error does not effect results of clients accessing server
+func ignorableError(err error) bool {
+	if err == nil {
+		return true
 	}
-
-	js, err := json.Marshal(resp)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+	if errors.Is(err, syscall.EPIPE) {
+		return true
+	}
+	if errors.Is(err, syscall.ECONNRESET) {
+		return true
 	}
 
-	js = append(js, '\n')
-	if _, err := w.Write(js); err != nil {
-		log.Printf("ERROR: Could not write response: %s", err)
+	switch err.Error() {
+	case "client disconnected":
```

Comment:
Any way that we could find a constant variable for this? Seems like a "magic" value.

Reply:
Any suggestions? This is an unexpected error from the `http` package.

```
./net/http/h2_bundle.go:	http2errClientDisconnected = errors.New("client disconnected")
```